source/MaterialXRuntime/Private/Commands/PvtCopyPrimCmd.cpp

```diff
@@ -86,60 +86,60 @@ RtPrim PvtCopyPrimCmd::createPrimCopy(const RtPrim& prim, const RtPath& parentPa
     }
 
     // Copy metadata that don't exists by default.
-    copyMetadata(PvtObject::hnd(prim)->asA<PvtObject>(), PvtObject::hnd(copy)->asA<PvtObject>());
+    copyMetadata(PvtObject::ptr(prim), PvtObject::ptr(copy));
 
     // Copy any inputs & outputs that don't exists by default.
     RtNodeGraph ng(prim);
     if (ng)
     {
-        // Special handling for nodegraphs which needs to call the nodegraph version of create input/outputs.
+        // Special handling for nodegraphs which needs to call the nodegraph version of createInput / createOutput.
         RtNodeGraph ngCopy(copy);
-        for (RtAttribute attr : ng.getInputs())
+        for (size_t i = 0; i < ng.prim()->numInputs(); ++i)
```

Comment:
Out of curiousity why isn't this for(RtInput inputPort : ng.getInputs()) same goes for getOutput below.